plugin/storage/cassandra/factory_test.go

```diff
@@ -87,3 +87,21 @@ func TestCassandraFactory(t *testing.T) {
 	_, err = f.CreateArchiveSpanWriter()
 	assert.NoError(t, err)
 }
+
+func TestCassandraFactoryWriterRateLimit(t *testing.T) {
+	logger, _ := testutils.NewLogger()
+	f := NewFactory()
+	v, command := config.Viperize(f.AddFlags)
+	command.ParseFlags([]string{
+		"--cassandra.writes-per-second=10",
+	})
+	f.InitFromViper(v)
+
+	// Use mock to avoid errors in unit test. See details in previous test case.
+	f.primaryConfig = &mockSessionBuilder{}
+	f.archiveConfig = nil
+	assert.NoError(t, f.Initialize(metrics.NullFactory, logger))
+
+	_, err := f.CreateSpanWriter()
+	assert.NoError(t, err)
```

Comment:
This test will pass regardless of the rate limiting setting. Maybe simply test for type, since any real test would involve sleeping and would take a long time to run.

Reply:
Fair point.